ddev/src/ddev/cli/validate/metadata.py

```diff
@@ -0,0 +1,300 @@
+
+
+def check_duplicate_values(current_check, line, row, header_name, duplicates, fail=None):
+    """Check if the given column value has been seen before.
+    Output a warning and return True if so.
+    """
+    if row[header_name] and row[header_name] not in duplicates:
+        duplicates.add(row[header_name])
+    elif row[header_name] != '':
+        message = f"{current_check.name}:{line} `{row[header_name]}` is a duplicate {header_name}\n"
+        if fail:
+            return (True, message)
+        else:
+            return (False, message)
+    return (False, '')
+
+
+@click.command(short_help='Validate `metadata.csv` files')
+@click.argument('integrations', nargs=-1)
+@click.option(
+    '--check-duplicates', is_flag=True, help='Output warnings if there are duplicate short names and descriptions'
+)
+@click.option('--show-warnings', '-w', is_flag=True, help='Show warnings in addition to failures')
+@click.pass_obj
+def metadata(app: Application, integrations: tuple[str, ...], check_duplicates: bool, show_warnings: bool):
+    """
+    Validate `metadata.csv` files
+
+    If `check` is specified, only the check will be validated, if check value is
+    'changed' will only apply to changed checks, an 'all' or empty `check` value
+    will validate all README files.
+    """
+    from ddev.cli.validate import metadata_utils
+
+    validation_tracker = app.create_validation_tracker('Metrics validation')
+
+    excluded = set(app.repo.config.get('/overrides/validate/metrics/exclude', []))
+    for current_check in app.repo.integrations.iter(integrations):
+        if current_check.name in excluded or not current_check.has_metrics:
+            continue
+
+        errors = False
+        error_message = ""
+        warning_message = ""
+
+        if current_check.name.startswith('datadog_checks_'):
+            continue
+
+        metric_prefix = current_check.manifest.get("/assets/integration/metrics/prefix", "")
+        metadata_file = current_check.metrics_file
+
+        # To make logging less verbose, common errors are counted for current check
+        metric_prefix_count: defaultdict[str, int] = defaultdict(int)
+        empty_count: defaultdict[str, int] = defaultdict(int)
+        empty_warning_count: defaultdict[str, int] = defaultdict(int)
+        duplicate_name_set: set = set()
+        duplicate_short_name_set: set = set()
+        duplicate_description_set: set = set()
+
+        metric_prefix_error_shown = False
+        if metadata_file.stat().st_size == 0:
+            errors = True
+
+            error_message += (
+                f"{current_check.name} metadata file is empty. This file needs the header row at the minimum.\n"
+            )
+
+        for line, row in read_metadata_rows(metadata_file):
+            # determine if number of columns is complete by checking for None values (DictReader populates missing columns with None https://docs.python.org/3.8/library/csv.html#csv.DictReader) # noqa
```

Comment:
You can split this into two comment lines to remove the ignore marker on the end